src/couch/src/couch_db.erl

```diff
@@ -369,8 +379,145 @@ get_full_doc_info(Db, Id) ->
 get_full_doc_infos(Db, Ids) ->
     couch_db_engine:open_docs(Db, Ids).
 
-purge_docs(#db{main_pid=Pid}, IdsRevs) ->
-    gen_server:call(Pid, {purge_docs, IdsRevs}).
+purge_docs(Db, IdRevs) ->
+    purge_docs(Db, IdRevs, []).
+
+-spec purge_docs(#db{}, [{UUId, Id, [Rev]}], [PurgeOption]) ->
+    {ok, [Reply]} when
+    UUId :: binary(),
+    Id :: binary(),
+    Rev :: {non_neg_integer(), binary()},
+    PurgeOption :: interactive_edit | replicated_changes,
+    Reply :: {ok, []} | {ok, [Rev]}.
+purge_docs(#db{main_pid = Pid} = Db, UUIDsIdsRevs, Options) ->
```

Comment:
I think this wouldn't work for PSE. We suppose to not have public access to `#db{}` record [see this commit](https://github.com/apache/couchdb/pull/495/commits/7b329b46ee7dccc549580e5f519023964771972c)

Reply:
couch_db and couch_db_updater are the only two modules that know about the record definition so they can introspect it.